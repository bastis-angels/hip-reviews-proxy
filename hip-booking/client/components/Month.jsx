import React from 'react'
import Week from './Week.jsx'
import moment from 'moment'

const Calendar = styled.div`
  ${props => (props.view === 'collapsed' || props.view === 'book') ? 'display: none;' : 'display: flex;'}
  flex-direction: column;
`
const PrevMonth = styled.span`
  padding-top: 4px;
  padding-bottom: 4px;
  ${props =>  props.current && 'color: #757575;'}
  display: block;
`
const NextMonth = styled.span`
  padding-top: 4px;
  padding-bottom: 4px;
  display: block;
`
const CurrentMonth = styled.span`
  padding-top: 4px;
  padding-bottom: 4px;
`
const MonthSelector = styled.div`
  display: flex;
  justify-content: space-around;
  font-size: 20px;
  font-weight: 500;
  border: 1px solid #ebebeb;
  border-top-width: 0;
  box-shadow: #ebebeb 0 1px 5px 0;
`
const DayLabelRow = styled.div`
  display: flex;
  border: 1px solid #ebebeb;
  border-top-width: 0;
  box-shadow: #ebebeb 0 1px 5px 0;
`
const DayLabel = styled.span`
  font-size: 1.2em;
  font-weight: 500;
  color: #333;
  background-color: #fff;
  width: 400;
  height: 36px;
  border-radius: 0;
  text-align: center;
  padding: 5px;
`
const Dates = styled.div`
  font-size: 1.2em;
  font-weight: 300;
  color: #333;
  background-color: #fff;
  border: 1px solid #ebebeb;
  width: 400;
  text-align: center;
`

export default class Month extends React.Component {
  constructor(props) {
    super(props);
    this.renderWeeks.bind(this);
  }

  renderWeeks() {
    let monthIndex = this.props.moment.month();
    let weekOfDate = this.props.moment.clone().startOf('month').day('Sunday');
    let weeks = [];
    let inProgress = true;

    while (inProgress) {
      weeks.push(<Week weekOfDate={weekOfDate.clone()} arrive={this.props.arrive} key={weekOfDate.format()} handleDayClick={this.props.handleDayClick} />);
      weekOfDate = weekOfDate.add(1, 'week');
      inProgress = monthIndex >= weekOfDate.month();
    }
    return weeks;
  }

  render() {
    return (
      <Calendar view={this.props.view}>
        <MonthSelector>
          <PrevMonth current={this.props.moment.month() === moment().month()} onClick={this.props.setPreviousMonth}> « </PrevMonth>
          <CurrentMonth> {this.props.moment.format('MMMM YYYY')} </CurrentMonth>
          <NextMonth onClick={this.props.setNextMonth}> » </NextMonth>
        </MonthSelector>
        <DayLabelRow>
          <DayLabel> S </DayLabel><DayLabel> M </DayLabel><DayLabel> T </DayLabel><DayLabel> W </DayLabel><DayLabel> T </DayLabel><DayLabel> F </DayLabel><DayLabel> S </DayLabel>
        </DayLabelRow>
        <Dates>
          {this.renderWeeks()}
        </Dates>
      </Calendar>
    )
  }
}  
