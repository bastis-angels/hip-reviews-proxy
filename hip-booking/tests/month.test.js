import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Month from '../client/components/Month.jsx'
import moment from 'moment'

it('renders month based on state.moment', () => {
  const month = shallow(<Month 
    view={'collapsed'}
    moment={moment()}
    arrive={'Select date'}
    handleDayClick={() => null} 
    setPreviousMonth={() => null} 
    setNextMonth={() => null}
  />);
  month.setState({moment: moment('2019-05-12')})
  expect(toJson(month)).toMatchSnapshot();
});