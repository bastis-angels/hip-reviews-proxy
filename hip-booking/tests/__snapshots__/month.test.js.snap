// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders month based on state.moment 1`] = `
<styled.div
  view="collapsed"
>
  <styled.div>
    <styled.span
      current={true}
      onClick={[Function]}
    >
       « 
    </styled.span>
    <styled.span>
       
      April 2019
       
    </styled.span>
    <styled.span
      onClick={[Function]}
    >
       » 
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
       S 
    </styled.span>
    <styled.span>
       M 
    </styled.span>
    <styled.span>
       T 
    </styled.span>
    <styled.span>
       W 
    </styled.span>
    <styled.span>
       T 
    </styled.span>
    <styled.span>
       F 
    </styled.span>
    <styled.span>
       S 
    </styled.span>
  </styled.div>
  <styled.div>
    <Week
      arrive="Select date"
      handleDayClick={[Function]}
      key="2019-03-31T00:00:00-07:00"
      weekOfDate={"2019-03-31T07:00:00.000Z"}
    />
    <Week
      arrive="Select date"
      handleDayClick={[Function]}
      key="2019-04-07T00:00:00-07:00"
      weekOfDate={"2019-04-07T07:00:00.000Z"}
    />
    <Week
      arrive="Select date"
      handleDayClick={[Function]}
      key="2019-04-14T00:00:00-07:00"
      weekOfDate={"2019-04-14T07:00:00.000Z"}
    />
    <Week
      arrive="Select date"
      handleDayClick={[Function]}
      key="2019-04-21T00:00:00-07:00"
      weekOfDate={"2019-04-21T07:00:00.000Z"}
    />
    <Week
      arrive="Select date"
      handleDayClick={[Function]}
      key="2019-04-28T00:00:00-07:00"
      weekOfDate={"2019-04-28T07:00:00.000Z"}
    />
  </styled.div>
</styled.div>
`;
